File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int mn = nums1[0];
        bool odd = false;
        for(int v : nums1){
            if(v < mn){
                mn = v;
            }

            if(v & 1){
                odd = true;
            }
        }

        if(mn & 1){
            return true;
        }
        return !odd;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int mn = nums1[0];
        boolean odd = false;
        for(int x : nums1){
            if(x < mn){
                mn = x;
            }

            if(x % 2 != 0){
                odd = true;
            }
        }
        if(mn % 2 != 0){
            return true;
        }
        return !odd;
    }
}